### PR TITLE
feat: add removePlugin api(resolve #3716)

### LINKED ISSUE #3716: [Request] provide removePlugin api
**Is your request related to a specific problem you're having?**
when I use vue, I add highlightjs plugin inside component, after that, each time when I mount this component, the highlightjs is added a plugin, if I mount  this component multiple times, highlightjs will be added multiple plugins.

**The solution you'd prefer / feature you'd like to see added...**
Provide a removePlugin api, so that I can invoke it before component unmount.

**Any alternative solutions you considered...**
I can't think of any other solutions, 

**Additional context...**
If there has workaround, I will also appreciate a lot.

### PR DESCRIPTION
**Pull request by faga295**
closed #3716 

### Checklist
- [x] Added markup tests, or they don't apply here because...
- [x] Updated the changelog at `CHANGES.md`

### CODE REVIEW

File: src/highlight.js [hunk 1]
```diff
@@ -724,7 +724,7 @@ const HLJS = function(hljs) {
     if (shouldNotHighlight(language)) return;
 
     fire("before:highlightElement",
-      { el: element, language: language });
```
Comment (joshgoebel) [inline src/highlight.js:727]: Which ES version was this short syntax introduced into, do you know?

File: src/highlight.js [hunk 2]
```diff
@@ -928,6 +928,16 @@ const HLJS = function(hljs) {
     plugins.push(plugin);
   }
 
+  /**
+   * @param {HLJSPlugin} plugin
+   */
+  function removePlugin(plugin) {
+    const index = plugins.indexOf(plugin);
+    if (index > -1) {
```
Comment (joshgoebel) [inline src/highlight.js:936]:
```suggestion
    if (index != -1) {
```

**Comment by joshgoebel** [comment]
Please remove all the purely linting changes.

**Comment by joshgoebel** [comment]
Actual code looks good, lets remove all the other stuff and we can get this merged.

**Comment by faga295** [comment]
thx, my  IDE do eslint fix on save, seems it didn't help

**Comment by faga295** [comment]
I withdraw lint change, please review it again.

**Review by joshgoebel — APPROVED** [review]

**Comment by joshgoebel** [comment]
Please add some short documentation to `docs/api` and then we can get this merged.

**Comment by faga295** [comment]
Should I add document for `addPlugin` as well?

**Comment by joshgoebel** [comment]
That'd be great.

File: docs/api.rst [hunk 3]
```diff
@@ -209,6 +209,24 @@ For example, if a new version suddenly had a serious bug (or breaking change) th
...
 * **In Debug Mode** all highlighting would stop and a JavaScript error would be thrown.
 
+addPlugin
+---------
+
+::
+
+  addPlugin(plugin)
+
+Add a plugin for hljs.See the :doc:`plugin api </plugin-api>` for more plugin information.
```
Comment (joshgoebel) [inline docs/api.rst:219]:
```suggestion
Add a plugin to this instance of Highlight.js.  See the :doc:`plugin api </plugin-api>` for additional plugin information.
```

File: docs/api.rst [hunk 4]
```diff
@@ -209,6 +209,24 @@ For example, if a new version suddenly had a serious bug (or breaking change) th
...
+
+removePlugin
+---------
+
+::
+
+  removePlugin(plugin)
+
+Remove a specified plugin for hljs.
```
Comment (joshgoebel) [inline docs/api.rst:228]:
```suggestion
Remove the specified plugin from this instance.  ``plugin`` must be exactly the same object that was passed to 
``addPlugin``.
```

File: docs/api.rst [hunk 5]
```diff
@@ -216,16 +216,15 @@ addPlugin
 
   addPlugin(plugin)
 
-Add a plugin for hljs.See the :doc:`plugin api </plugin-api>` for more plugin information.
-
+Add a plugin to this instance of Highlight.js.  See the :doc:`plugin api </plugin-api>` for additional plugin information.
 removePlugin
 ---------
```
Comment (joshgoebel) [inline docs/api.rst:221]:
```suggestion
Add a plugin to this instance of Highlight.js.  See the :doc:`plugin api </plugin-api>` for additional plugin information.


removePlugin
---------
```

**Review by joshgoebel — APPROVED** [review]

**Pull request merged**